### PR TITLE
docs(typehead): add demo for using with forms #3748

### LINKED ISSUE #3748: docs(typehead): add demo for using with forms
Add demo for using with template-driven form (related to #3342)

### PR DESCRIPTION
**Pull request by IraErshova**
Add demo for using with template-driven forms

Close #3748

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [ ] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [ ] built and tested the changes locally.
 - [ ] added/updated API documentation.
 - [ ] added/updated demos.

### CODE REVIEW

File: demo/src/app/components/+typeahead/demos/reactive-form/reactive-form.ts [hunk 1]
```diff
@@ -4,10 +4,10 @@ import { FormControl, FormGroup } from '@angular/forms';
...
 
 @Component({
-  selector: 'demo-typeahead-forms',
-  templateUrl: './in-form.html'
+  selector: 'demo-typeahead-reactive-form',
+  templateUrl: './reactive-form.html'
 })
-export class DemoTypeaheadFormsComponent {
+export class DemoTypeaheadReactiveFormComponent {
   stateCtrl: FormControl = new FormControl();
```
Comment (IlyaSurmay) [inline demo/src/app/components/+typeahead/demos/reactive-form/reactive-form.ts]: remove type annotations if a field has a default value

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

**Comment by YevheniiaMazur** [comment]
Tested, looks good

File: demo/src/app/components/+typeahead/typeahead-section.list.ts [hunk 2]
```diff
@@ -56,12 +57,27 @@ export const demoComponentContent: ContentSection[] = [
...
+        outlet: DemoTypeaheadFormComponent
+      },
       {
         title: 'Reactive forms',
         anchor: 'reactive-forms',
-        component: require('!!raw-loader?lang=typescript!./demos/in-form/in-form.ts'),
-        html: require('!!raw-loader?lang=markup!./demos/in-form/in-form.html'),
-        outlet: DemoTypeaheadFormsComponent
+        description: `
+          <p>Typeahead can be used in reactive forms. Keep in mind that value of <code>ngModel</code> is string</p>
```
Comment (valorkin) [inline demo/src/app/components/+typeahead/typeahead-section.list.ts]: reactive forms doesn't use ngModel

Comment (valorkin) [reply demo/src/app/components/+typeahead/typeahead-section.list.ts]: please change wording

**Review by valorkin — CHANGES_REQUESTED** [review]

**Review by valorkin — APPROVED** [review]

**Pull request merged**